content/blog/2024/12/zephyr-how-to-use-psram/index.md

```diff
@@ -0,0 +1,741 @@
+I (184) boot: DROM segment: paddr=00080000h, vaddr=3c060000h, size=16160h ( 90464) map
+I (192) boot: IROM segment: paddr=00020000h, vaddr=42000000h, size=549C2h (346562) map
+I (211) soc_random: Disabling RNG early entropy source
+I (211) boot: Disabling glitch detection
+I (211) boot: Jumping to the main image...
+I (212) octal_psram: vendor id    : 0x0d (AP)
+I (216) octal_psram: dev id       : 0x02 (generation 3)
+I (221) octal_psram: density      : 0x03 (64 Mbit)
+I (225) octal_psram: good-die     : 0x01 (Pass)
+I (230) octal_psram: Latency      : 0x01 (Fixed)
+I (234) octal_psram: VCC          : 0x01 (3V)
+I (238) octal_psram: SRF          : 0x01 (Fast Refresh)
+I (243) octal_psram: BurstType    : 0x01 (Hybrid Wrap)
+I (248) octal_psram: BurstLen     : 0x01 (32 Byte)
+I (252) octal_psram: Readlatency  : 0x02 (10 cycles@Fixed)
+I (258) octal_psram: DriveStrength: 0x00 (1/1)
+I (262) MSPI Timing: PSRAM timing tuning index: 5
+I (266) esp_psram: Found 8MB PSRAM device
+I (270) esp_psram: Speed: 80MHz
+I (686) esp_psram: SPI SRAM memory test OK
+I (714) heap_runtime: ESP heap runtime init at 0x3fca8788 size 259 kB.
+
+*** Booting Zephyr OS build v4.0.0-2029-g6e0e5591a4fb ***
+uart:~$ wifi scan
+Scan requested
+
+Num  | SSID                             (len) | Chan (Band)   | RSSI | Security        | BSSID             | MFP
+1    | Soares                           6     | 11   (2.4GHz) | -63  | WPA2-PSK        | 90:0A:62:42:A5:BF | Disable
+...
+uart:~$ wifi connect --key-mgmt 1 --ssid <SSID> --passphrase <PASSPHRASE>
+Connection requested
+Connected
+[00:01:31.902,000] <inf> net_dhcpv4: Received: 192.168.15.2
+uart:~$ net ping 192.168.15.17
+PING 192.168.15.17
+28 bytes from 192.168.15.17 to 192.168.15.2: icmp_seq=1 ttl=64 time=231 ms
+28 bytes from 192.168.15.17 to 192.168.15.2: icmp_seq=2 ttl=64 time=6 ms
+28 bytes from 192.168.15.17 to 192.168.15.2: icmp_seq=3 ttl=64 time=338 ms
+uart:~$ net allocs
+Network memory allocations
+memory		Status	Pool	Function alloc -> freed
+0x3c083920	 free	   RX	eth_esp32_rx():135 -> processing_data():178
+0x3c0836a0	 free	   TX	dhcpv4_create_message():287 -> ethernet_send():762
+0x3c082894	 free	TDATA	ethernet_fill_header():532 -> ethernet_remove_l2_header():61
+```
+
+Examining the output from both sessions, we can confirm that in the first case, network stack allocations were made using memory from internal RAM -- addresses 0x3fca4af0, 0x3fca4870, and 0x3fca3a2c. We can also confirm that in the second case, network stack allocations were taking memory from PSRAM -- addresses 0x3c083920, 0x3c0836a0, and 0x3c082894 -- thus proving that enabling `ESP32_WIFI_NET_ALLOC_SPIRAM` avoids allocating memory from the precious internal SRAM and takes advantage of PSRAM.
+
+## Final Thoughts
+
+Throughout this article, we explored three different strategies for utilizing PSRAM: using it as a heap for dynamic memory allocation, integrating it as a section in the memory map, and moving code from flash to PSRAM.
```

Comment:
In the intro and here, you talk about the three strategies as this:

- using it as a heap for dynamic memory allocation
- integrating it as a section in the memory map
- moving code from flash to PSRAM

However, the actual article sections are called:

- Using PSRAM for Dynamic Memory Allocation
- Adding PSRAM to Your ESP32 Memory Map
- Running Code from PSRAM Instead of Flash

Suggest that you use the same names for these strategies both in text and section titles.